Server/src/server/gameCenter/models/game/Player.java

```diff
@@ -91,6 +92,35 @@ public void addCardToDeck(ServerCard card) throws LogicException {
         return drawnCards;
     }
 
+    public ServerCard[] getCardsFromDeckNoDuplicate(int cardsToDraw, ServerCard card) {
+        ServerCard[] drawnCards = new ServerCard[cardsToDraw];
+        Random RNGenerator = new Random();
+        int infiniteRedrawFailSafe = 10;
+
+		for(int i = 0; i < cardsToDraw; i++) {
+            if (!deck.getCards().isEmpty()) {
+                int index = RNGenerator.nextInt(deck.getCards().size());
```

Comment:
Did you check how endpoints work with indexing.
e.g. if size is 10,  can next int return 10 (or are endpoints exclusive?).  

If nextInt is inclusive of 10 then thats an index error waiting to happen

Reply:
0 is inclusive upper bound is exclusive